keps/sig-node/3675-resource-plugin-manager/README.md

```diff
@@ -0,0 +1,1229 @@
+of abilities while stopping the continued increasing complexity within the core of
+the Kubelet.
+
+Users would like to be able to address the following use cases:
+
+* The ability to allow vendors to release vendor-specific managers for their hardware
+  and provide an interface while not having vendor-specific code within Kubelet.
+* Differentiate between types of cores and memory. <br>
+  Note - Dynamic resouce allocation does this with regular resources today.  We seek to
+  extend this ability.
+* Diffentiate between different configurations of cores and memory, for instance cores
+  designated as performance versus those designated as efficiency cores 
+* Have custom plugins to optimize for particular types of workloads.  These plugins
+  may be built for performance, power reduction, cpu savings, et cetera.  <br>
+  Note:  Currently, there are very limited sets of available topology policies.  Every
+  new policy must be approved and be lockstep with the Kuberenetes release process.
+* Be able to hot-plug and test new resource managers.  
+* Be able to remove some of the complexity with current setup and, over time, reduce
+  the amount of code contained within Kubelet.  Instead, build a library with specific
+  needs.
+* Have a faster path to desired changes, without potentially impacting the core of 
+  Kubernetes with every policy change. <br>
+  Note that current solutions have been cropping up to allow for resource management
+  outside the core Kubelet.  THese solutions require turning off the Kubelet functionality 
+  and overriding current Kubelet allocation.  We should provide a path otherwise.
+* Be able to get information on the pods on the node without having to contact the
+  API server, which may not have updated information. 
+* Be able to do research, with minimum toil, on new policies and resource management strategies
+
+This design will also use the already tried and true gRPC, which is used for many other
+pluggable components within Kubernetes.  The hope is that it will also, as a side effect, allow a 
+path to simplify the Kubelet as it is today and move the complexity into external plugins.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+#### Alpha Goals
+
+1.	Provide Initial infrastructure to support CCI Plugins and k8s Deployments requiring CCI Drivers
+2.	Support seamless k8s start ( no plugins required) through the CCI CPU management policy
+3.	Support minimal set of plugin-less Pod-deployment through CPU Manager – equivalent to best-effort QoS (if sufficient time – static handling of burstable and guaranteed QoS)
+4.	Resource sets support cpu and memory
+5.	Consistent state handling between CCI Driver and CPU Manager
+6.	Support proper fail mechanisms of Pods if CCI driver is not available
+7.	Continue cluster operation if CCI driver is not available 
```

Comment:
Would it make sense to add scheduling approach as an Alpha goal?

Reply:
CCI Drivers come into play after scheduling decisions, we will add clarification in the doc for that

Reply:
Is it an architectural choice to avoid changing scheduling and only concentrate on the node operations? If so, please state it clearly in the document.